extensions/indexes/range/src/main/java/org/exist/xquery/modules/range/RangeQueryRewriter.java

```diff
@@ -138,13 +138,17 @@ protected static Lookup rewrite(Expression expression, NodePath path) throws XPa
             InternalFunctionCall fcall = (InternalFunctionCall) expression;
             Function function = fcall.getFunction();
             if (function instanceof Lookup) {
-                if (function.isCalledAs("matches")) {
-                    eqArgs.add(function.getArgument(0));
-                    eqArgs.add(function.getArgument(1));
-                    Lookup func = Lookup.create(function.getContext(), RangeIndex.Operator.MATCH, path);
-                    func.setArguments(eqArgs);
-                    return func;
+                final RangeIndex.Operator operator = RangeIndex.Operator.getByName(function.getName().getLocalPart());
+
+                if(function==null){
```

Comment:
The null check is too late here. Should go further up.

Reply:
done